torch/csrc/autograd/python_variable.cpp

```diff
@@ -1267,7 +1256,7 @@ PyObject *THPVariable_pynew(PyTypeObject *type, PyObject *args, PyObject *kwargs
   HANDLE_TH_ERRORS
   TORCH_CHECK(type != &THPVariableType, "Cannot directly construct _TensorBase; subclass it and then construct that");
   jit::tracer::warn("torch.Tensor", jit::tracer::WARN_CONSTRUCTOR);
-  auto tensor = torch::utils::legacy_tensor_ctor(torch::tensors::get_default_dispatch_key(), torch::tensors::get_default_scalar_type(), args, kwargs);
+  auto tensor = torch::utils::legacy_tensor_ctor(type, torch::tensors::get_default_dispatch_key(), torch::tensors::get_default_scalar_type(), args, kwargs);
```

Comment:
Do I read correctly that this constructor does not accept the "requires_grad" flag today at all? And thus can't be used to do `Foo(torch.empty(2), requires_grad=True)`?
In that case, I think I like the idea of raising an error if the input requires_grad and not in no_grad mode. We can even recommend to use a custom Function if they want to make this construction differentiable.

Reply:
Based in your chat comments I was planning to readd the requires grad field, seemed like a good idea.

Reply:
So the final thing would be:
- You have a requires_grad flag you can use to create a leaf that requires grad
- The given Tensor can never require gradients.

Reply:
One thing that gives me pause about forcing the given Tensor to not require gradients is that it is a bit tiresome for AOTAutograd when you get a requires_grad input; we actually do want to turn these into leaves so that we can compute gradients only to them. But I guess it is not too bad; it looks like `Tracer(x.detach(), requires_grad=x.requires_grad)